### ISSUE TITLE
Ringtones disappear from sounds and tones

### ISSUE THREAD

**Issue by jv2007** [nextcloud/android#5039]
### Actual behavior

Ringtones disapper from under sounds and tones as soon as an autoupload completes successfully from the nextcloud app. The sounds come back if the phone is re started, and they disapper again after any auto upload. This causes phone not to ring at all because there are no sounds/tones available under the sounds options

### Expected behaviour
- The sounds should not disappear after every automatic upload to the server. 
 
### Steps to reproduce
1. Turn on auto upload for camera photos
2. Click a photo, and wait for auto upload to complete
3. Check under sounds and tones, you should see none.

### Environment data
Android version: 10

Device model: Oneplus GM1901

Stock or customized system: Oxygen OS 10.0.1.GM57AA

Nextcloud app version: 3.9.2

Nextcloud server version: 17

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by GarciST** [comment]
Verified. I can reproduce the same behaviour in my OnePlus 7t with 10.0.7.HD65BA Oxygen OS version. 

Ringtones and notification tones are missing when Next loud app syncs.

Quick fix to recover ringtones and notification tones: reboot system (but it is not a valid solution)

**Comment by AndyScherzinger** [comment]
Cc @tobiasKaminsky since I think this is a false conclusion. We as an app don't touch anything near your ringtones so I'd rather think you hit an OxygenOS bug we can do nothing about.

What do you think @tobiasKaminsky?

**Comment by jv2007** [comment]
The issue still persists, the ringtones disappear everytime there is an automatic upload from nextcloud app

Regards,

Jivesh Vashist

________________________________

**Comment by jv2007** [comment]
Any questions or updates?

Regards,

Jivesh Vashist

________________________________

**Comment by luckyrat** [comment]
I appreciate it's tough to identify whether it's OxygenOS, Android 10 or NextCloud that needs changing but I can confirm that it's been happening on my device since last month's update to Android 10 (previously Android 9).

I've had to disable photo uploads which is obviously a large part of what makes the NextCloud app useful so it would be good to at least identify what the cause of the problem is so we can supply relevant information/logs to the right people.

I've got hundreds of apps installed and this very specific behaviour of the NextCloud app is the only thing that triggers the problem so even if it's an OxygenOS bug, I hope that someone familiar with the relevant NextCloud code could hypothesise what might be going on.

I've only been able to reproduce the problem with InstantUploads. The Notes sync app and automatic contacts backup work fine.

Android version: 10 (SDK 29)
Device model: Oneplus A6003
Stock or customized system: Oxygen OS 10.3.1
Nextcloud app version: 3.10.1
Nextcloud server version: 17.0.3

**Comment by AndyScherzinger** [comment]
Looping in @tobiasKaminsky @ezaquarii @ArisuOngaku since you folks might have an idea if/why such a behavior might get triggered. 🤞

**Comment by ezaquarii** [comment]
This is super weird. Is it happening only on OnePlus devices?

**Comment by ArisuOngaku** [comment]
It might be related to scoped storage (pure guess); someone who can reproduce the issue please send a log file (stripped out of sensitive data).

**Comment by luckyrat** [comment]
I'm happy to provide some logs but I can't work out how to find them. Please can someone direct me to the menu option to copy the logs or some instructions to help me find them?

**Comment by ArisuOngaku** [comment]
@luckyrat A quick look in the code got me to think it was removed from the main app; @AndyScherzinger can you confirm? In this case I would ask you to reproduce this with the **dev** version of the app, then get the logs in the "More" category of the settings.

**Comment by AndyScherzinger** [comment]
@ArisuOngaku Logging is only visible in debug builds and ***dev*** 

https://github.com/nextcloud/android/blob/master/src/main/java/com/owncloud/android/ui/activity/SettingsActivity.java#L359-L375

**Comment by nitefood** [comment]
I confirm I am experiencing the same behavior, in similar conditions. Oneplus 6 running OxygenOS 10.3.2 (latest available update). I started experiencing this problem as soon as Android was updated from v9 to v10 on my device.
I may also add this piece of information: I have customized notifications for Whatsapp and Telegram (both private and group messages), using built-in notification sounds (stock OxygenOS sounds). As soon as the ringtone problem occurs, all these app-specific notifications get also set to strange, non-existent values. For instance, I get a number like "2617" where I used to see "Ding" as the notification sound, or "2622" where I had "Light". Of course this means I hear no sound at all, unless I manually re-set the notification sounds.
At the same time, system-wide notification sounds like SMS tone, which I have set to an audio file I stored in internal memory, or all those apps configured with the default system-wide notification sound, don't get changed. So apparently only app-specific notifications get changed (and the phone ringtone)

**Comment by ArisuOngaku** [comment]
@luckyrat the easiest way to obtain the logs is to use the **dev** version of the app. Then find it in settings > more > logs

**Comment by ArisuOngaku** [comment]
@nitefood thanks for your input. If this is possible, please share the logs of the app by reproducing the bug with the **dev** version of the app then grabing the file from settings > more > logs. Don't forget to strip out sensitive data if any (instance address, usernames...).

**Comment by masx14** [comment]
> This is super weird. Is it happening only on OnePlus devices?

No, not only my Oneplus 7 Pro is affected,  but also the Google Pixel 3A of my girlfriend. 

With the oneplus 7 I am able to restart the phone every night and only upload the photos when the phone is charging in the evening, but with the Pixel 3A I cannot do that unfortunately.

**Comment by tobiasKaminsky** [comment]
I have the same with Android 10 and LineageOS.
Not sure what/why/if we are causing this…

**Comment by tobiasKaminsky** [comment]
I am debugging this right now.
One observation is that custom ringtones/notification sounds seem not be affected, e.g. I set it to one mp3 within my NC and it is still there, although all other built-in sounds are removed.

**Comment by tobiasKaminsky** [comment]
https://issuetracker.google.com/issues/139255464

**Comment by OlivierS1** [comment]
I have been fighting this issue since I purchased my OnePlus 7T about a month ago.
Latest update to Oxygen OS 10.0.9.HD65AA last night and problem still persist.

**Comment by tobiasKaminsky** [comment]
I really think that this is a bug out of scope of any app.
Imagine that any malicious app could change notification/ringtone sound, this would be a big problem…

**Comment by nateyoung** [comment]
I have this same problem on 2 phones in my family. Both are Motorola One Hyper running Android 10. After a photo is auto uploaded, ringtones/notifications show up as numbers instead of names, except custom mp3's. Manually uploading a photo works fine. Have to disable auto upload for now. I did change the default destination upload folder if that matters.

kernel 4.14.117-perf+
#1 Wed Feb 12 08:28:43 CST 2020

**Comment by tobiasKaminsky** [comment]
What I do not get is the connection to AutoUpload, as this is mainly doing the same like upload
- uploads a file
- shows a notification

So if the notification would be the problem, it should also occur on manual file upload.

Then only difference is that auto upload is started as as background job…

**Comment by tobiasKaminsky** [comment]
I set my tones to a custom one and since then it works for me.
Can you verify this?

**Comment by masx14** [comment]
For my Oneplus this doesn’t work.

**Comment by nateyoung** [comment]
This seems to work for me. I rebooted phone and verify my normal ringtones/notifications are set properly. Take a picture, it auto-uploads, I get notifications in my status bar while it's happening (but no sound notification - not sure if this is normal), then check notification settings - they all show numbers instead of the real names. Trying to change them shows a list with nothing in it. Rebooting makes the list populate w/ all the default notification names and all is back to normal (until the next auto-upload). Again, manually uploading doesn't seem to trigger this.

If I set notifications to a mp3 from the internet, all of the above is still true, except there are no numbers listed, and trying to select another ringtone shows an empty list again except for the mp3 I previously set.

So it seems like wherever the storage for the default system notifications is is getting messed up, or some cache, or media scan, or something. Let me know if you have other experiments to try or want to gather logs somehow.

**Comment by tobiasKaminsky** [comment]
I added our findings to the google issue tracker: https://issuetracker.google.com/issues/139255464

**Comment by tobiasKaminsky** [comment]
@nateyoung can you go to app settings -> notifications and disable "upload channel" and see if the sound is still vanishing?
If yes, then please disable all other channels and try it again.

Thanks for testing!

**Comment by tobiasKaminsky** [comment]
wow, did not expected this :S
On Monday I'll try to all sound related calls in notifications and provide a test version.

**Comment by nateyoung** [comment]
A couple of other datapoints:

-I happened to be trying to recreate this and managed to create a conflict upload. I resolved it manually and the picture uploaded with notifications, but this didn't break the notification sound selection as above.

-After recreating the failure mode in another test, I rebooted and immediately went in to notification settings. They were still broken for a few seconds but then something refreshed and they were all back. I haven't figured out how to manually refresh this though. It's like the default system notification storage loses an index or cache or something.

**Comment by tobiasKaminsky** [comment]
Thanks for you sharing your findings.

> It's like the default system notification storage loses an index or cache or something.

Well, this then sounds even more like a bug on Android, which accidently triggered by us.

**Comment by nateyoung** [comment]
I did figure out a workaround: using ES File Explorer, I copied all tones from Phone Storage -> product -> media -> audio -> alarms/notifications/ringtones to Internal Storage -> Alarms/Notifications/Ringtones. Then Android will show duplicates when selecting a tone, but after a Nextcloud auto upload, only the new copies are left behind. Select from these, and they persist after auto upload, reboot, etc. Duplicates come back after reboot but disappear after the first auto upload.

**Comment by tobiasKaminsky** [comment]
@nateyoung thanks for sharing this workaround.
So it proves that only the built-in notifications are affected…

**Comment by elektrikfisch** [comment]
@nateyoung Thanks for posting that workaround. I'll try it as well.

Checking in here with the same issue. OnePlus 7T running OxygenOS 10.0.9.HD65AA. Bootloader unlocked and rooted with Magisk. No further system modifications other than some Magisk modules that were installed after I first observed this issue.

I've had to keep auto uploads turned off, but I can confirm that this issue presented itself yesterday upon a MANUAL upload.

Side note, Nextcloud-connected apps are working fine without triggering this issue: DAVx5, Notes, and PhoneTrack

Please let me know how I can help.

**Comment by elektrikfisch** [comment]
OnePlus 7T, US unlocked, OxygenOS 10.0.9.HD65AA, bootloader unlocked, rooted with Magisk

I downloaded the Dev build 20200415 from F-Droid, and performed the following actions:

- 2020-04-29T09:01:03.559-0400 Log in, with TOTP
- 2020-04-29T09:03:20.829-0400 Upload an 8 KB .csv file
- Check Android Sound & vibration settings: All ringtones OK
- 2020-04-29T09:03:56.619-0400 Upload a 252 KB .jpg file
- Check Android Sound & vibration settings: All ringtones MISSING
- Save Nextcloud Dev debug log

[logs-redacted.txt](https://github.com/nextcloud/android/files/4551863/logs-redacted.txt)

**Comment by elektrikfisch** [comment]
This appears to only be related to uploading image files? Something to do with thumbnail generation perhaps?

Following from above:
- 2020-04-29T09:28:18.802-0400 Upload the same 8 KB .csv file, now to root folder on Nextcloud
- Check Sound & vibration settings: All ringtones OK
- 2020-04-29T09:28:50.817-0400 Upload 70 B .xml file to same folder
- Check Sound & vibration settings: All ringtones OK
- 2020-04-29T09:29:51.422-0400 Upload 7.4 MB .apk file to same folder
- All ringtones OK
- 2020-04-29T09:30:23.278-0400 Upload 159 KB .jpg file to same folder
- All ringtones MISSING

[logs2-redacted.txt](https://github.com/nextcloud/android/files/4551982/logs2-redacted.txt)

Apologies for multiple comments in a row.

**Comment by tobiasKaminsky** [comment]
@elektrikfisch nice finding!

But we are using for this an official Android API:
https://github.com/nextcloud/android/blob/7eb0f66079e2fc91d948dec4d84eca5b689b3521/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java#L1737-L1743

https://developer.android.com/reference/android/content/Intent#ACTION_MEDIA_SCANNER_SCAN_FILE

But it seems that this deprecated.
Let me quickly fix this.

**Comment by tobiasKaminsky** [comment]
Hm, not that easy to implement as there is no real doc how to use this…

Anyway: https://github.com/nextcloud/android/pull/5969 simply removes media scanner on Android 10.
Soon there will be an APK which you can install.
Can you test it with this?

Downside might be that the media scanner is not triggered and thus the new image is not shown in gallery. But this is first to test if the ringtones remain…

**Comment by elektrikfisch** [comment]
@tobiasKaminsky I can install and test with that! Where can I find the new APK? Does it get attached to that pull request after it's done building?

And which gallery are you referring to? Android's systemwide gallery? I think that if a user uploads an image to Nextcloud, that image would already be in their gallery anyway, wouldn't it?

**Comment by tobiasKaminsky** [comment]
> Does it get attached to that pull request after it's done building?

Yes.



> Android's systemwide gallery?

Yes



> I think that if a user uploads an image to Nextcloud, that image would already be in their gallery anyway, wouldn't it?

If you choose to upload & move file into NC, it might be that the file is in gallery, but the location is not updated and thus it would be still shown in DCIM/Camera.

**Comment by elektrikfisch** [comment]
> If you choose to upload & move file into NC, it might be that the file is in gallery, but the location is not updated and thus it would be still shown in DCIM/Camera.

I see, that makes sense.

I'll stand by for the build to finish. Thanks for doing this!

**Comment by tobiasKaminsky** [comment]
Here is the apk to test:
https://github.com/nextcloud/android/pull/5969#issuecomment-621353022

I fear that you have to disable/remove the official version so that you can make sure that it is working.

**Comment by elektrikfisch** [comment]
@tobiasKaminsky Thanks for that change. At first attempt, #5969 seems to have solved the issue! I'll play with it some more this evening and tomorrow and see if I can break it.

**Comment by tobiasKaminsky** [comment]
@elektrikfisch is the uploaded file then shown in correct location in gallery app?
(as we now do not trigger media scanner)

**Comment by elektrikfisch** [comment]
@tobiasKaminsky When I upload with the "Keep file in source folder" option, the Gallery doesn't change. When I upload with the options "Delete file from source folder" or "Move file to #SITENAME# folder", the image immediately disappears from the Gallery app.

I've always uploaded using the keep in source folder option, so I'm not sure exactly what the normal behavior is for the other options on the current stable release.

**Comment by tobiasKaminsky** [comment]
> When I upload with the options "Delete file from source folder" or "Move file to #SITENAME# folder", the image immediately disappears from the Gallery app.

And is the image then appearing in the folderName of NC?
So if you upload it to /instantUpload/2020/04 there should be a folder "04" with the uploaded image in gallery

**Comment by elektrikfisch** [comment]
@tobiasKaminsky when I use the Delete or Move options, the photo disappears from the Gallery app. It does not re-appear in the Gallery app under a different location. Gallery has a list of "App Collections" which is populated by Download, Discord, and Telegram. Nextcloud is not present.

But this is interesting... I reinstalled the stable release to see how it behaves differently. The uploaded photo still does not display in the Gallery if I use the Move or Delete options. However, this issue also does not appear to be triggered when I use those options.

**Comment by elektrikfisch** [comment]
@tobiasKaminsky So I cleaned up to run another test scenario. Uninstalled QA and stable. Installed dev, logged in, and rebooted (to clear the missing ringtone state).

- Take 3 new photos
- Upload first photo with Delete option. Ringtones OK. Photo missing from Gallery and Camera app.
- Upload second photo with Move option. Ringtones OK. Photo missing from Gallery and Camera app. 
- Upload third photo with Keep option. Ringtones MISSING. Photo remains in Gallery and Camera app.

To maybe find a different view into the gallery, I opened up Solid Explorer and opened Photos, and found a folder titled #USER#@#HOST#? The second photo (uploaded with Move option) is within that folder. Actual path to file is /storage/emulated/0/Android/media/com.nextcloud.android.beta/nextcloud/#USER#@#HOST#/IMG_20200430_073549.jpg . I'm not sure if this is reading from the Android gallery, or if Solid Explorer finds photos independently. Is there anything else I should try?

Log attached: [logs3-redacted.txt](https://github.com/nextcloud/android/files/4558036/logs3-redacted.txt)

**Comment by tobiasKaminsky** [comment]
Thanks for testing.
I finally also found an emulator where I can reproduce it.
Sadly, when I move a file while uploading it is not shown in gallery, so I have to find a way to replace the old media scanner for newer Android versions…

**Comment by axheli** [comment]
@tobiasKaminsky  Hello, 

Let me know if you need me to do any testing i have a OnePlus 7 Pro on Android 10 Im currently Running 3.12.0 RC1 and still having the issue for now i stoped auto uploads but i even did a manual upload and still the same issue

**Comment by skeletor-spills** [comment]
Confirming @axheli .  Manual upload on OnePlus 7T OxygenOS latest build. Both manual and auto-upload to local server remove built-in ringtone sounds from system.

It looks like the CI/CD is holding up the fix in #5969.  Anything I can do there?

**Comment by axheli** [comment]
@skeletor-spills What dose that mean CI/CD ?

**Comment by skeletor-spills** [comment]
@axheli, I should have mentioned the developer/maintainer on the CI/CD thing (Continuous Integration / Continuous Delivery).

Some of the automatic tests for gplay are failing according to the referenced issue. 

@tobiasKaminsky , is this because it needs to rescan media?

BTW Thanks a ton for maintaining some of this huge project and working through this issue.

**Comment by axheli** [comment]
@skeletor-Thank you very much :) 

@tobiasKaminsky Yes, at agree with @skeletor-spills thank you so much :)

**Comment by elektrikfisch** [comment]
@tobiasKaminsky @AndyScherzinger 

My ringtones are breaking again. I haven't had a chance to experiment to *prove* that it's Nextcloud or to put together any repro steps or logs, but I'm guessing that since this only started happening in the last couple of days, and 3.14.0 includes more changes to media uploads, that maybe we have a regression?

Currently using Google Play build v 3.14.0

I'll have some time this afternoon (GMT-5) to play with it.

**Comment by OlivierS1** [comment]
I have noticed this too and the behavior still remains the same.
I loose my ringtone when Auto uploads transfer files.
In my case, another app broke after it was updated, for possibly other reasons, but this happened right around an update of my phone to Oxygen OS 10.0.14.
That was about 3 weeks ago.

**Comment by AndyScherzinger** [comment]
I am not aware of any changes with respect to auto upload as-in upload. The auto upload change shipping with 3.14.0 (done by myself) is the filtering for detecting files (as in ignore certain patterns) so the change would only lead to less auto detected folders (as in notification to active auto upload for them) and a shortened auto upload screen list (since some folders will now be ignored by default: as-in music folders with just a single cover.jpg)

cc @tobiasKaminsky

**Comment by axheli** [comment]
I had this issue before but not having any issues on v 3.14

**Comment by fyah** [comment]
hello
same behavior, happy to find this thread..  3.14 from play store as well
let me know if I can provide any log

**Comment by jv2007** [comment]
Thank you for reopening the issue.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by tobiasKaminsky** [comment]
No idea :/
It was really hard to track the previous bug down.
Problem was the old media scanner system.
But we changed nothing in this regard…

I remember that this happened also when manual uploading a file with "move or delete" option.
Can you try this again, so that we know that it is NC?
And please also specify Android version.

**Comment by OlivierS1** [comment]
I just tried again.
I did the following:
1) Took a picture.
2) Went quickly under "Sound & vibration"  => "Phone ringtone and vibration":
 List of ringtone sounds are available.
3) Nextcloud icon pops up at the top of the display indicating transfer is in progress.
4) Return to previous menu and go under "Phone ringtone and vibration". List of ringtone sounds is now empty.

This is Android 10 on OnePlus 7T.

**Comment by fyah** [comment]
Hello,
on my side i revoked the Nextcloud app storage permission and rebooted. No problem since yesterday nigth.

This is Android 10 (october patch) Nokia 8.3

**Comment by elektrikfisch** [comment]
> Hello,
> on my side i revoked the Nextcloud app storage permission and rebooted. No problem since yesterday nigth.
> 
> This is Android 10 (october patch) Nokia 8.3

To clarify, have you re-enabled Storage permissions for the Nextcloud app without any further issues? Or have you left it revoked, and thus have limited app functionality?

**Comment by fyah** [comment]
> To clarify, have you re-enabled Storage permissions for the Nextcloud app without any further issues? Or have you left it revoked, and thus have limited app functionality?

I have made the opposite test of Olivier. It means that now my nextcloud app miss the functionality I need the most (i.e auto upload)
Leaving the permisison revoked and seeing that the problem do not occur anymore is a way to proove that the issue comes from nextcloud app, and that it is triggered by the autoupload process

I wonder what kind of logs we could get to findout what the issue is.

**Comment by axheli** [comment]
I'm on 3.14.1 RC2 and i dont have any upload issues i take a picture uploads right away.

**Comment by elektrikfisch** [comment]
@tobiasKaminsky 

Can confirm this issue is reproducible the same as before :\
I uninstalled NC the other day and haven't had an issue since. Today, I installed the latest build from F-Droid.

1. Installed Nextcloud from F-Droid
2. Log into NC instance
3. Verified that all expected notification sounds are available
4. Grant Storage permission to Nextcloud app
5. Manually upload a file (specifically a 9 KB .png file this time) using the delete source file option.
6. Observe that no sounds are available in the main notification sounds settings, and all custom notification sounds set in various apps are broken and their titles replaced by numbers.

Nextcloud Android app version 3.14.0
OnePlus 7T running latest stock OTA Oxygen OS 10.0.15.HD65AA, systemless rooted using Magisk

**Comment by elektrikfisch** [comment]
> I'm on 3.14.1 RC2 and i dont have any upload issues i take a picture uploads right away.

What version of Android? As far as I know, this has only been reproducible on 10

**Comment by axheli** [comment]
> > I'm on 3.14.1 RC2 and i dont have any upload issues i take a picture uploads right away.
> 
> What version of Android? As far as I know, this has only been reproducible on 10

Im on Oneplus 8 Pro Oxygen OS 11.0.1.1.IN11AA. I had the same issue a while back on my Oneplus 7 Pro and Oneplus 8 Pro on Android 10. I created a ticket with oneplus a while back https://forums.oneplus.com/threads/missing-sounds-on-phone-third-party-app-issue.1227270/#post-21761498 they said they where going to fix it but i guess they never did. I know this only happens on Oneplus phones

**Comment by axheli** [comment]
Also heres the ticket i created her when i had the same issue 

Nextcloud Android 10 Issue #6061

**Comment by elektrikfisch** [comment]
The issue is reproducible using the latest Nextcloud Dev build on F-Droid, 20201120

I went back and reinstalled the previous stable build on F-Droid, version 3.13.1, and the issue is NOT reproducible. There are no issues with notification sounds on version 3.13.1

Again, OnePlus 7T running Oxygen OS 10.0.15.HD65AA

**Comment by elektrikfisch** [comment]
@axheli Is that a preview build for Android 11? And the issue is NOT happening there?

**Comment by axheli** [comment]
I understand that is the way the program is doing its file scan to upload the content to the server. Something to do with the Oxygen OS and the way NC app is doing its upload that it dose not like.

**Comment by elektrikfisch** [comment]
But to confirm, you're saying that this issue does not happen in 11.0.1.1.IN11AA?

**Comment by fyah** [comment]
Not only oxygen os. I am affected as well on a Nokia 8 running Android 10. It is pretty much stock Android (Android one)

**Comment by axheli** [comment]
Nope dose not happen on 11.0.1.1.IN11AA. I'm telling you is something wired idk why this keeps on happening to these phones. I'd also recommend putting in a ticket with Oneplus and reference the ticket i have open. Nextcloud should deffinelty look at this

Please see below notes this might be the issue @tobiasKaminsky  

I currently have a Oneplus 7 Pro and due to the way the Nextcloud app that the below info can explaine is casing issues that all my notification on the phone go away. and this is due to Nextcloud and Android 10

For anyone wondering about this still: The issue in my case is an app using a deprecated Android API to scan media files. In my case the Nextcloud client app was using the following code:

Code:
public static void triggerMediaScan(String path) {
if (path != null) {
Intent intent = new Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE);
intent.setData(Uri.fromFile(new File(path)));
MainApp.getAppContext().sendBroadcast(intent);
}
}
Apparently, Android 10 does not like this. I've sent it to the support crew here in hopes that they can fix it as well. Most likely if you are having this issue, there's an app using similarly structured code to scan for media files in this manner which is deprecated in Android 10 (which is what the current version of OxygenOS is based on).

My advice is to find the app causing the issue and also report the problem to the app developer or customer support.

**Comment by axheli** [comment]
@elektrikfisch Yes, I completely agree with you. My question is what in the world is the difference between Android 10 or 11. and another comment if you download your own notifications sounds and not use the default sounds from OnePlus they don't disappear.

**Comment by elektrikfisch** [comment]
> @elektrikfisch Yes, I completely agree with you. My question is what in the world is the difference between Android 10 or 11. and another comment if you download your own notifications sounds and not use the default sounds from OnePlus they don't disappear.

Previously, my main notification sound was a custom one, and it would suffer the same fate as the stock sounds.

Also, this could be an indication that OnePlus did indeed fix what you reported... it just took them a whole major release lol

**Comment by elektrikfisch** [comment]
@tobiasKaminsky Can you re-upload the test apk that was built for #6984 ? And maybe another apk from another build shortly before that? Could be that commit b01fdbab1b8c21fd66646ce85b7c7172ccd847de caused this regression.

**Comment by katerasrael** [comment]
I can confirm the bug for Oneplus Nord, Oxygen OS 10.5.9. AC01BA and nextcloud app 3.14.0

**Comment by tobiasKaminsky** [comment]
> Could be that commit [b01fdba](https://github.com/nextcloud/android/commit/b01fdbab1b8c21fd66646ce85b7c7172ccd847de) caused this regression.

This only skips media scanning if the file is not there.
As far as I understood this whole (could it only test once on a phone), this happens when media scanner runs.

I can build one from 3.14. without this.

**Comment by tobiasKaminsky** [comment]
There is an APK file in PR #7504, which you can install in parallel to your existing Nextcloud app.
It would be great if you can verify the fix.

**Comment by fyah** [comment]
Installed the QA version 16992.apk.
configured auto upload for camera folder
checked ringtones OK
took a photo, seen the upload notification, upload completed
checked ringtones OK

**Comment by elektrikfisch** [comment]
Tested 16992.apk from #7504 using manual upload with delete source file. Ringtones are OK.

OnePlus 7T Oxygen OS 10.0.15.HD65AA

**Comment by katerasrael** [comment]
Tested the QA16992, no issues, working as expected!

Oneplus Nord, Oxygen OS 10.5.9

**Comment by RobertJansen1** [comment]
i installed the QA APK and now auto upload does not break my ringtones anymore on my just bought oneplus nord, great work!

**Comment by DaveSpe** [comment]
Issue still exists on latest version of LineageOS. 
Not an issue on Stock android.

**Comment by serdarevica** [comment]
Hi QA apk is not available any longer? I'd there final/alternative fix for this?

Thanks

**Comment by DaveSpe** [comment]
> Hi,
> 
> QA apk is not available any longer?
> Is there final/alternative fix for this?
> 
> Thanks

I would also like to know if this is being pushed as an update to production.

**Comment by serdarevica** [comment]
> > Hi,
> > QA apk is not available any longer?
> > Is there final/alternative fix for this?
> > Thanks
> 
> I would also like to know if this is being pushed as an update to production.

I guess no since there is no response. 😐

**Comment by AndoKarim** [comment]
Tested the QA App that is back for people who want it. Check the PR mentionned above. 

Worked perfectly with me. 
OnePlus 7 Oxygen OS 10.3.7.GM57AA

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky see comments above, your fix seems to solve the issue (even though I don't quite understand the fix or why that would solve it)

**Comment by serdarevica** [comment]
> Tested the QA App that is back for people who want it. Check the PR mentionned above.
> 
> Worked perfectly with me.
> OnePlus 7 Oxygen OS 10.3.7.GM57AA

If i install QA app now, am i going to get notification in playStore about new/updated final release with this fix included?

**Comment by DaveSpe** [comment]
I'm not interested in running two instances of the app. When will this fix be pushed tp.production?

**Comment by AndyScherzinger** [comment]
> If i install QA app now, am i going to get notification in playStore about new/updated final release with this fix included?

If you still have the original play store version installed too, yes. The QA build (QA=Quality Assurance) is a build for testing purposes only so this is only for anyone to validate that the fix works which helps with developing the app and is also a contribution to this app's further development.

> I'm not interested in running two instances of the app. When will this fix be pushed tp.production?

Your choice of course. What you are actually saying though is: that you are not willing to help test the fix (provided for testing with the QA build) but of course ask for when others will do the quality assurance and release it for you. So to answer your question: When people who are willing to contribute some time and test the fix report back that it is working.

**Comment by fyah** [comment]
> Edit: I get the solution, path being detected but empty just seems strange

The && !TextUtils.isEmpty(path) statements makes it so that if the path is empty, then the code goes out of the section dedicated to Android 10 and higher (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q).

To have the same level of protection against empty pathes (which I assume was the reason for commit b01fdbab1b8c21fd66646ce85b7c7172ccd847de was introduced) I guess the test for empty path shall be put one level bellow ?

**Comment by PyBonnetainNesterenko** [comment]
LineageOS 17.1 here, on OnePlus 3T, and NC 3.14.0 from F-Droid (no access to Google Play). Same problem. Cannot help on the code, but if you need testing tell me how.

**Comment by martoni35** [comment]
Hello,
I use an oneplus 6T global with lineageOS 17.1 with the latest lineageOS image.
I configured the sound for notification, alarm and phone. Everything is ok for notification and phone, but the alarm works with notification sound. And, when i received a call, it is not the call sound but notification sound. I looked the configuration sound for alarm and it was numbers for the sounds.. I rebooted it and the names for sound are OK. 
Few hours ago, because nextcloud upload if i understand, the number for sounds back...no good sounds for all...

Few mounts ago, i had the same problem : https://gitlab.com/LineageOS/issues/android/-/issues/2002
My nextcloud client is V3.14.1
You can see screenshots about numbers when the phone failed.
The rest of the device works fine also
martoni

**Comment by tobiasKaminsky** [comment]
There is an APK file in PR #7691, which you can install in parallel to your existing Nextcloud app.
It would be great if you can verify the fix.

**Comment by fyah** [comment]
hello,
after installing the APK 7691, I can connect, but then the app crash
I can never get to ue the app, it always crash at startup

**Comment by Sargastic** [comment]
> It would be great if you can verify the fix.

Installed. Just to be safe, I uninstalled the current NC app, to avoid any interference wrt ringtones. Will keep you posted.

**Comment by Sargastic** [comment]
Same problem as fyah met. Install ok, connect to the server (with an application token) ok. The app starts, displays one folder, starts some scan and crashes.

**Comment by skeletor-spills** [comment]
Confirming #7691 QA build APK crashes after log in. Will test ringtones and sounds when new build is up. Thanks.

**Comment by Fruchuxs** [comment]
In which case is the path null or empty? Maybe this is a workaround until the fix is pushed.

**Comment by AndyScherzinger** [comment]
The fix that has been successfully tested by some reporters here has been merged for the 3.14.2 release. Which @tobiasKaminsky can/will take care of after his vacation (I can't do official releases, sorry 😢)

**Comment by AndyScherzinger** [comment]
If anyone still wants to test the fix the latest QA build would be https://github.com/nextcloud/android/pull/7714#issuecomment-749474027 which also has other fixes for 3.14.x.

I am closing this issue for now and flag it for 3.14.2

**Comment by Sargastic** [comment]
> If anyone still wants to test the fix the latest QA build would be [#7714 (comment)](https://github.com/nextcloud/android/pull/7714#issuecomment-749474027) which also has other fixes for 3.14.x.

AFAIK (Lineage 17.1 on OnePlus 3T) this build solves the ringtones problem.

**Comment by DaveSpe** [comment]
This started happening again in a recent version of the app.
